src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -368,6 +367,23 @@ void GcsServer::InitGcsPlacementGroupManager(const GcsInitData &gcs_init_data) {
   rpc_server_.RegisterService(*placement_group_info_service_);
 }
 
+std::string GcsServer::StorageType() const {
+  if (RayConfig::instance().gcs_storage() == "memory") {
+    if (!config_.redis_address.empty()) {
```

Comment:
it's wired, why we use redis instead when gcs storage is memory? shouldn't we crash here?

Reply:
I'm treating `RayConfig::instance(). gcs_storage()` as a feature flag for `using inmemory KV by default`, instead of a configuration option to always use inmemory KV. Otherwise, users would need to specify both `RAY_gcs_storage=redis` and `RAY_REDIS_ADDRESS=...` to use external Redis. And we will never be able to remove the `gcs_storage` flag. I understand we did discuss about supporting different storage types via `gcs_storage`. It may be ok to require user to specify `RAY_gcs_storage=redis` but it seems unnecessary.

Reply:
I see. should we deprecate this flag and pass external-address from flag then? that sounds like a final solution. I mean one flay should be good enough.

Reply:
Yeah, after the GCS migration, we would deprecate all 3 feature flags.